pkgs/games/steam/chrootenv.nix

```diff
@@ -178,6 +178,12 @@ in buildFHSUserEnv rec {
   '';
 
   profile = ''
+    # Workaround for issue #44254 (Steam cannot connect to friends network)
+    # https://github.com/NixOS/nixpkgs/issues/44254
+    test -z ''${TZ+x} &&
+    new_TZ="$(readlink -f /etc/localtime | grep -o '(?<=/zoneinfo/).*$' -P)" &&
+    export TZ="$new_TZ"
```

Comment:
Is there a reason for using an intermediate variable instead of directly exporting `TZ` as the result of the `readlink | grep` expansion?

Reply:
Yes. This way TZ is left alone if the `grep` fails.

Reply:
Oh indeed. Maybe add in a comment to warn future code readers not to do what I was thinking of? :)